iati/tests/test_resources.py

```diff
@@ -50,193 +177,492 @@ class TestResourceFolders(object):
+        (iati.version.STANDARD_VERSION_ANY, iati.resources.PATH_VERSION_INDEPENDENT)
     ])
     @pytest.mark.latest_version('2.02')
-    def test_folder_name_for_version(self, version, expected_version_foldername):
-        """Check that expected components are present within folder paths."""
-        path = iati.resources.folder_name_for_version(version)
+    def test_version_folder_name_generation_known(self, version, expected_version_foldername):
+        """Check that the correct folder name is returned for known version numbers."""
+        folder_name = iati.resources.folder_name_for_version(version)
 
-        assert expected_version_foldername == path
+        assert expected_version_foldername == folder_name
 
-    @pytest.mark.parametrize('version', [
-        '1.00',
-        1,
-        1.01,  # A version must be specified as a string
-        'string'
-    ])
-    def test_folder_name_for_version_invalid_version(self, version):
-        """Check that an invalid version of the Standard raises a ValueError exception."""
+    def test_version_folder_name_generation_unknown(self, std_ver_all_mixedinst_valid_unknown):
+        """Check that a ValueError is raised when trying to create a folder name for an unknown version."""
         with pytest.raises(ValueError):
-            iati.resources.folder_name_for_version(version)
+            iati.resources.folder_name_for_version(std_ver_all_mixedinst_valid_unknown)
 
-    @pytest.mark.parametrize('path_component', [
-        'resources',
-        'standard'
-    ])
-    def test_folder_path_for_version(self, std_ver_minor_mixedinst_valid_fullsupport, path_component):
-        """Check that expected components are present within folder paths."""
-        path = iati.resources.folder_path_for_version(std_ver_minor_mixedinst_valid_fullsupport)
-        assert path_component in path
+    def test_folder_name_for_version_valueerr(self, std_ver_all_uninst_valueerr):
+        """Check that an version of the Standard of the correct type, but an incorrect value raises a ValueError."""
+        with pytest.raises(ValueError):
+            iati.resources.folder_name_for_version(std_ver_all_uninst_valueerr)
 
-    def test_get_test_data_paths_in_folder(self):
-        """Check that test data is being found in specified subfolders.
+    def test_folder_name_for_version_typeerr(self, std_ver_all_uninst_typeerr):
+        """Check that an version of the Standard of the correct type, but an incorrect value raises a TypeError."""
+        with pytest.raises(TypeError):
+            iati.resources.folder_name_for_version(std_ver_all_uninst_typeerr)
 
-        Todo:
-            Deal with multiple versions.
 
-        """
-        paths = iati.tests.resources.get_test_data_paths_in_folder('ssot-activity-xml-fail', '2.02')
+@pytest.mark.new_tests
+class TestResoucePathCreationEntireStandard(object):
+    """A container for tests relating to generating entire filepaths for any part of the Standard."""
```

Comment:
Docstring should be clarified to better indicate how it differs from other similarly named classes.